securesystemslib/storage.py

```diff
@@ -0,0 +1,248 @@
+
+    <Arguments>
+      fileobj:
+        The file-like object to be stored.
+
+      filepath:
+        The full path to the location where 'fileobj' will be stored.
+
+    <Exceptions>
+      securesystemslib.exceptions.StorageError, if the file can not be stored.
+
+    <Returns>
+      None
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+  @abc.abstractmethod
+  def getsize(self, filepath):
+    """
+    <Purpose>
+      Retrieve the size, in bytes, of the file at 'filepath'.
+
+    <Arguments>
+      filepath:
+        The full path to the file.
+
+    <Exceptions>
+      securesystemslib.exceptions.StorageError, if the file does not exist or is
+      not accessible.
+
+    <Returns>
+      The size in bytes of the file at 'filepath'.
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+  @abc.abstractmethod
+  def create_folder(self, filepath):
+    """
+    <Purpose>
+      Create a folder at filepath and ensure all intermediate components of the
+      path exist.
+
+    <Arguments>
+      filepath:
+        The full path of the folder to be created.
+
+    <Exceptions>
+      securesystemslib.exceptions.StorageError, if the folder can not be
+      created.
+
+    <Returns>
+      None
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+  @abc.abstractmethod
+  def list_folder(self, filepath):
+    """
+    <Purpose>
+      List the contents of the folder at 'filepath'.
+
+    <Arguments>
+      filepath:
+        The full path of the folder to be listed.
+
+    <Exceptions>
+      securesystemslib.exceptions.StorageError, if the file does not exist or is
+      not accessible.
+
+    <Returns>
+      A list containing the names of the files in the folder. May be an empty
+      list.
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+
+
+
+class FilesystemBackend(StorageBackendInterface):
+  """
+  <Purpose>
+    A concrete implementation of StorageBackendInterface which interacts with
+    local filesystems using Python standard library functions.
+  """
+
+  class GetFile(object):
+    # Implementing get() as a function with the @contextmanager decorator
+    # doesn't allow us to cleanly capture exceptions thrown by the underlying
+    # implementation and bubble up our generic
+    # securesystemslib.exceptions.StorageError, therefore we implement get as
+    # a class and also assign the class to the 'get' attribute of the parent
+    # FilesystemBackend class.
+
+    def __init__(self, filepath):
+      self.filepath = filepath
+
+
+    def __enter__(self):
+      try:
+        self.file_object = open(self.filepath, 'rb')
+        return self.file_object
+      except (FileNotFoundError, IOError):
+        raise securesystemslib.exceptions.StorageError(
+            "Can't open %s" % self.filepath)
+
+
+    def __exit__(self, exc_type, exc_val, traceback):
+      self.file_object.close()
+
+
+
+# Map our class ContextManager implementation to the function expected of the
```

Comment:
Nit: Mismatched indentation? Ignore me if it's a style choice 🙂

Reply:
Thanks for catching that. It wasn't intentional. Fixed in a force-push